server/controller/user.js

```diff
@@ -0,0 +1,143 @@
+const User = require('../models/User')
+const jwt = require('../helpers/jwt.js')
+const bcrypt = require('bcryptjs')
+const {OAuth2Client} = require('google-auth-library');
+const client = new OAuth2Client(`${process.env.GOOGLESIGNINID}`)
+
+class UserController {
+    static register(req, res, next) {
+        let newUser = {
+            first_name: req.body.first_name,
+            last_name: req.body.last_name,
+            email: req.body.email,
+            password: req.body.password,
+            avatar : "https://image.flaticon.com/icons/png/512/78/78373.png"
+        }
+        User.create(newUser)
+            .then((gotData) => {
+                res.json(gotData)
+            })
+            .catch(next)
+
+    }
+
+    static findByemail(req,res,next){
+        User.findOne({email : req.params.email})
+        .then((gotData)=>{
+            if(gotData){
+                res.json(gotData)
+            } else {
+                throw {code : 404, message: "not found"}
+            }
+        })
+        .catch(next)
+    }
+    static login(req, res, next) {
+        console.log("masuk login");
+        
+        User.findOne({
+                email: req.body.email
+            })
+            .then((gotData) => {
+                if (!gotData) {
+                    throw ({
+                        code: 404,
+                        message: " Email Not Found"
```

Comment:
jangan kasih tahu kalo emailnya tidak ditemukan, saat proses login infokan saja: email/password is wrong